### PR TITLE
Disable laser on PWM or pulse width <= 0

### PR DESCRIPTION
**Pull request by mzieg**
- setting PWM 0% disables laser
- update MacOS build to latest Homebrew

*NEEDS TESTING*

### CODE REVIEW

File: WasatchVCPPLib/WasatchVCPPLib/Spectrometer.cpp [hunk 1]
```diff
@@ -207,60 +207,69 @@ bool WasatchVCPP::Spectrometer::setLaserPowerPerc(float percent)
...
     logger.debug("set_laser_power_perc: range (0, 100), requested %.2f, applying %.2f", percent, value);
 
-    value = float(max(0.f, min(100.f, value)));
-    // If full power(and allowed), disable modulationand exit
+    // If zero, disable laser
+    if (value <= 0)
+        return setLaserEnable(false);
+
+    // If full power (and allowed), disable modulation and exit
     if (value >= 100) {
```
Comment (jkleinwp) [inline WasatchVCPPLib/WasatchVCPPLib/Spectrometer.cpp:219]: I don't think this value could ever be >100 due to #211

File: WasatchVCPPLib/WasatchVCPPLib/Spectrometer.cpp [hunk 2]
```diff
@@ -207,60 +207,69 @@ bool WasatchVCPP::Spectrometer::setLaserPowerPerc(float percent)
...
             setModEnable(false);
             return true;
         }
     }
+
     int period_us = laserPowerHighResolution ? 1000 : 100;
-    int width_us = int((value * period_us / 100.0)); // note that value is in range(0, 100) not (0, 1)
+    int width_us = int((value * period_us / 100.0)); // note that value is in range (0, 100) not (0, 1)
+
+    if (width_us <= 0)
```
Comment (jkleinwp) [inline WasatchVCPPLib/WasatchVCPPLib/Spectrometer.cpp:236]: This would never be less than 0

**Review by jkleinwp — APPROVED** [review]

**Comment by mzieg** [comment]
Agreed with your comments. I just like preventing future bugs, and clamps are cheap :-)  (Arguably increases readability, in case someone hadn't read the earlier lines.) Basically I like validating input data at every step in a pipeline.

**Comment by jkleinwp** [comment]
> Agreed with your comments. I just like preventing future bugs, and clamps are cheap :-) (Arguably increases readability, in case someone hadn't read the earlier lines.) Basically I like validating input data at every step in a pipeline.

Fair point!

**Pull request merged**